newIDE/app/src/EventsFunctionsExtensionsLoader/Storage/LocalEventsFunctionsExtensionOpener.js

```diff
@@ -23,30 +23,28 @@ const readJSONFile = (filepath: string): Promise<Object> => {
 
 export default class LocalEventsFunctionsExtensionOpener {
   static chooseEventsFunctionExtensionFile = (): Promise<?string> => {
-    return new Promise((resolve, reject) => {
-      if (!dialog) return reject('Not supported');
+    if (!dialog) return Promise.reject('Not supported');
+    const browserWindow = electron.remote.getCurrentWindow();
 
-      const browserWindow = electron.remote.getCurrentWindow();
-      dialog.showOpenDialog(
-        browserWindow,
-        {
-          title: 'Import an extension in the project',
-          properties: ['openFile'],
-          message: 'Choose an extension file to import (.json file)',
-          filters: [
-            {
-              name: 'GDevelop 5 "events based" extension',
-              extensions: ['json'],
-            },
-          ],
-        },
-        paths => {
-          if (!paths || !paths.length) return resolve(null);
-
-          return resolve(paths[0]);
-        }
-      );
-    });
+    return dialog
+      .showOpenDialog(browserWindow, {
+        title: 'Import an extension in the project',
+        properties: ['openFile'],
+        message: 'Choose an extension file to import (.json file)',
+        filters: [
+          {
+            name: 'GDevelop 5 "events based" extension',
+            extensions: ['json'],
+          },
+        ],
+      })
+      .then(({ filePaths }) => {
+        if (!filePaths || !filePaths.length) return null;
+        return filePaths[0];
+      })
+      .catch(err => {
```

Comment:
Let's remove all the `catch`s, unless there is a case in which showOpenDialog can throw an exception - but I don't think so.
Exceptions are signaling a serious problem: having one there means that there was something that was not handled - in JavaScript this can even be a syntax error. We do *not* want to catch these. This is called "exception swallowing" and this hides the issue to the caller functions.
If an exception was launched, we're in very serious trouble, and we should let the exception propagate and crash the app (with a proper stacktrace in development that will let us find what was the issue). Static type checkers like Flow are helping us to ensure that, in theory, nothing bad will happen.

Reply:
Right.. Removing the catches.